src/main/java/cohen/worldcitysearch/WorldCitiesList.java

```diff
@@ -0,0 +1,76 @@
+package cohen.worldcitysearch;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
+public class WorldCitiesList
+{
+    private final ArrayList<WorldCity> worldCities;
+
+    public WorldCitiesList() throws IOException
+    {
+        this.worldCities = new ArrayList<>();
+
+        File csvData = new File("src/main/resources/worldcities.csv");
+        CSVParser parser = CSVParser.parse(csvData, Charset.defaultCharset(), CSVFormat.RFC4180);
+
+        List<CSVRecord> cities = parser.getRecords();
+        int totalRecords = cities.size();
+
+        for (int i = 1; i < totalRecords; i++)
+        {
+            String city = cities.get(i).get(1);
+            String lat = cities.get(i).get(2);
+            String lon = cities.get(i).get(3);
+            WorldCity worldCity = new WorldCity(
+                    city,
+                    Double.parseDouble(lat),
+                    Double.parseDouble(lon)
+            );
+            worldCities.add(worldCity);
+        }
+    }
+
+    public ArrayList<WorldCity> getList()
+    {
+        return worldCities;
+    }
+
+    public WorldCity getClosestCity(WorldCity requestCity)
+    {
+        //what about kd-trees?? bounding boxes? to make this faster. Although seems like overall
+        //no way to get past O(n) time in worst case scenario
+
+        WorldCity closest = worldCities.get(0);
+
+        for (int i = 1; i < worldCities.size(); i++)
+        {
+            double currentDist = distance(requestCity, closest);
```

Comment:
Calculating distance is an expensive operation. 
Save this value and reuse it.

Reply:
Doesn't this change at each iteration of the loop?

Reply:
`currentDist` does not change every loop. 
It only changes when the closest city changes.